### PR TITLE
Allow to use unix socket as redis host

### PR DESCRIPTION
**Pull request by deees**
Reuse host field for unix socket.  Not a nicest solution, bet at least we don't introduce additional field to already crowded form 😄 

Related: https://github.com/piwik/plugin-QueuedTracking/issues/37

screen shot 2016-11-11 at 2 07 24 pm

### CODE REVIEW

**Comment by mattab** [comment]
if/when merging, let's also merge https://github.com/piwik/component-cache/pull/14/files

**Comment by mattab** [comment]
Thanks for the PR @deees !

File: Queue/Backend/Redis.php [hunk 1]
```diff
@@ -233,7 +233,7 @@ private function connectIfNeeded()
     protected function connect()
     {
         $this->redis = new \Redis();
-        $success = $this->redis->connect($this->host, $this->port, $this->timeout, null, 100);
+        $success = $this->redis->pconnect($this->host, $this->port, $this->timeout, null, 100);
```
Comment (tsteur) [inline Queue/Backend/Redis.php]: Is this change needed? Doesn't this open a persistent connection? It seems to be a feature of phpredis? but I am actually not sure whether we are still using PHPredis. I think we have replaced it with credis. Couldn't find a command pconnect

Comment (deees) [inline Queue/Backend/Redis.php]: ah, yes, it's unrelated to this pull request, will remove it.

You are right, `pconnect` is to open persistent connection using phpredis (phpredis is used in QueuedTracking if sentinel is not enabled). `pconnect` is quite useful for high traffic websites. We were using it before unix socket support, because `connect` leaves a lot of tcp connections in `TIME-WAIT` state (every request makes a new connection and does not close it at the end of request).

**Comment by tsteur** [comment]
Looks good to merge for me. @mattab any thoughts re wording? Otherwise we could merge and release a new version

**Comment by mattab** [comment]
:+1: (haven't released the new version)
